scenario_player/runner.py

```diff
@@ -2,42 +2,172 @@
+from raiden.constants import UINT256_MAX
+from raiden.network.proxies.proxy_manager import ProxyManager
+from raiden.network.proxies.token import Token
+from raiden.network.proxies.token_network_registry import TokenNetworkRegistry
+from raiden.network.proxies.user_deposit import UserDeposit
+from raiden.network.rpc.client import EthTransfer, JSONRPCClient
+from raiden.settings import RAIDEN_CONTRACT_VERSION
+from raiden.utils.typing import TransactionHash, UserDepositAddress
 from scenario_player.constants import (
     API_URL_TOKEN_NETWORK_ADDRESS,
-    API_URL_TOKENS,
     NODE_ACCOUNT_BALANCE_FUND,
     NODE_ACCOUNT_BALANCE_MIN,
     OWN_ACCOUNT_BALANCE_MIN,
     RUN_NUMBER_FILENAME,
 )
 from scenario_player.definition import ScenarioDefinition
-from scenario_player.exceptions import ScenarioError, TokenRegistrationError
+from scenario_player.exceptions import ScenarioError
 from scenario_player.exceptions.legacy import TokenNetworkDiscoveryTimeout
-from scenario_player.services.utils.interface import ServiceInterface
-from scenario_player.utils import TimeOutHTTPAdapter, get_udc_and_token, wait_for_txs
-from scenario_player.utils.token import Token, UserDepositContract
+from scenario_player.node_support import NodeController, RaidenReleaseKeeper
+from scenario_player.utils import TimeOutHTTPAdapter, wait_for_txs
+from scenario_player.utils.configuration.settings import SettingsConfig, UDCSettingsConfig
+from scenario_player.utils.token import (
+    load_token_configuration_from_file,
+    token_maybe_mint,
+    token_registry_maybe_register_token,
+    userdeposit_maybe_deposit,
+    userdeposit_maybe_increase_allowance,
+)
 
 if TYPE_CHECKING:
     from scenario_player.tasks.base import Task, TaskState
 
 log = structlog.get_logger(__name__)
 
 
+def get_udc_and_corresponding_token(
+    udc_settings: UDCSettingsConfig, settings: SettingsConfig, proxy_manager: ProxyManager
+) -> Tuple[UserDeposit, Token]:
+    """ Return contract proxies for the UserDepositContract and associated token """
+    assert udc_settings.enable
+
+    udc_address = udc_settings.address
+
+    if udc_address is None:
+        chain_id = settings.chain_id
+        assert chain_id, "Missing configuration, either set udc_address or the chain_id"
+
+        contracts = get_contracts_deployment_info(chain_id, version=RAIDEN_CONTRACT_VERSION)
+
+        msg = (
+            f"invalid chain_id, {chain_id} is not available for version {RAIDEN_CONTRACT_VERSION}"
+        )
+        assert contracts, msg
+
+        udc_address = contracts["contracts"][CONTRACT_USER_DEPOSIT]["address"]
+
+    user_deposit_proxy = proxy_manager.user_deposit(
+        UserDepositAddress(to_bytes(udc_address)), "latest"
+    )
+
+    token_address = user_deposit_proxy.token_address("latest")
+    token_proxy = proxy_manager.token(token_address, "latest")
+
+    return user_deposit_proxy, token_proxy
+
+
+def determine_run_number(scenario_dir: Path) -> int:
```

Comment:
This corresponds to the [`ScenarioRunner.determine_run_number`](https://github.com/raiden-network/scenario-player/blob/f208ed6cb4f713c1efc921b10458365bcf918b5d/scenario_player/runner.py#L121-L135), this doesn't need to be a method since it does not required internal state. Moved it out and simplified the signature/code.